### DISCUSSION TITLE
Magic Link Expiration - Invite User

### DISCUSSION THREAD

**Discussion by nkeat12** [supabase/supabase#13527] [General]
**End Goal**
To send emails to people (via sendgrid) that contain a magic link that when clicked, automatically:
1.  Creates a user in our system
2. Logs them in
3. redirects them to a unique url. 

This should be seamless/not noticeable to a user at all, to them they are clicking the link and landing on a landing page. 

----

_We currently have this working (sending via the default supabase template) by calling the signInWithOtp method from our app._ 

**However, we need a way to**
* Significantly increase the OTP expiration time past the 86400 seconds / 1 day limit that is imposed within the supabase dashboard (say 30 days or whatever, or ideally never expire).
* Send this link via Sendgrid/3rd party transactional email provider. 

How can we achieve this? If its not possible with any out of the box methods, is there a recommended alternative path (e.g some middleware that then refreshes the token or creates a new one when the link is clicked?

**Comment by 1Ghasthunter1** [comment]
Any updates on this?
